src/components/study/programming-language-section.tsx

```diff
@@ -0,0 +1,146 @@
+'use client'
+
+import React, { useEffect } from 'react'
+import { useQuery } from '@/common'
+import { useSession } from '@/hooks/session'
+import { InputError } from '@/components/errors'
+import { getOrgsWithLanguagesAction } from '@/server/actions/org.actions'
+import { StudyProposalFormValues } from '@/app/[orgSlug]/study/request/study-proposal-form-schema'
+import { Divider, Grid, Paper, Radio, Stack, Text, Title } from '@mantine/core'
+import { UseFormReturnType } from '@mantine/form'
+import { getOrgBySlug, isOrgAdmin, OrgWithLanguages } from '@/lib/types'
+
+type Props = {
+    form: UseFormReturnType<StudyProposalFormValues>
+}
+
+export type ProgrammingLanguageUiState = {
+    orgName: string
+    isSingleLanguage: boolean
+    options: ReadonlyArray<'R' | 'PYTHON'>
+}
+
+export const deriveProgrammingLanguageUiState = ({
+    org,
+    isAdmin,
+}: {
+    org: OrgWithLanguages | null
+    isAdmin: boolean
+}): ProgrammingLanguageUiState => {
+    if (!org) {
+        return {
+            orgName: '',
+            isSingleLanguage: false,
+            options: [],
+        }
+    }
+
+    if (isAdmin) {
+        // Admin reviewers can always see both R and Python. We deliberately do not auto-select
+        // a language for admins; they must choose explicitly.
+        // TODO: In a later stage the general admin handling of test base images and supported languages should be refactored.
+        return {
```

Comment:
where is this specified?  why would admins for one org be able to force a different org to support a language?